src/validate/validate_global_state_expression.test.ts

```diff
@@ -0,0 +1,69 @@
+import {createExpression} from '..';
+import {findGlobalStateExpressionKeys} from './validate_global_state_expression';
+
+test('findGlobalStateExpressionKeys extracts all global state keys', () => {
+    const expression = [
+        'all',
+        ['>=', ['get', 'max_power'], ['global-state', 'ev_preferred_power']],
+        [
+            'case',
+            [
+                'in',
+                null,
+                ['global-state', 'ev_preferred_cpo']
+            ],
+            true,
+            [
+                'in',
+                ['get', 'brand'],
+                ['global-state', 'ev_preferred_cpo']
+            ]
+        ],
+        [
+            'case',
+            [
+                'in',
+                null,
+                ['global-state', 'ev_preferred_connector_type']
+            ],
+            true,
+            [
+                'in',
+                ['get', 'connector_type'],
+                ['global-state', 'ev_preferred_connector_type']
+            ]
+        ],
+        [
+            'case',
+            [
+                'in',
+                null,
+                ['global-state', 'ev_preferred_emsp']
+            ],
+            true,
+            [
+                'in',
+                ['get', 'payment_type'],
+                ['global-state', 'ev_preferred_emsp']
+            ]
+        ]
+    ];
+
+    const parsedExpression = createExpression(expression);
+
+    if (parsedExpression.result === 'error') {
```

Comment:
Logic is not recommended in tests.
Better to use `expect` instead.

Reply:
Changed to `fail`. Check is needed, otherwise TS is complaining.